python_backend/api/v1/endpoints/finetune.py

```
@@ -0,0 +1,205 @@
+        
+        result = await finetune_service.generate_with_finetuned_model_only(
+            input_text=request.text
+        )
+        
+        return FinetuneResponse(**result)
+        
+    except ValueError as ve:
+        raise HTTPException(
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail=str(ve)
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)